src/unix/async.c

```diff
@@ -138,9 +136,11 @@ static void uv__async_io(uv_loop_t* loop, uv__io_t* w, unsigned int events) {
     QUEUE_REMOVE(q);
     QUEUE_INSERT_TAIL(&loop->async_handles, q);
 
-    if (0 == uv__async_spin(h))
+    if (!(cmpxchgi(&h->pending, ASYNC_PENDING, 0) & ASYNC_PENDING))
       continue;  /* Not pending. */
 
+    cmpxchgi(&h->pending, ASYNC_PENDING | ASYNC_BUSY, ASYNC_BUSY); /*  Clear Pending */
```

Comment:
Long line, please keep it <= 80 columns. You should check that the return value is what you expect it to be.